notes/bilinear_pairings.tex

```diff
@@ -0,0 +1,666 @@
+\documentclass{article}
+
+% Some notation
+\newcommand{\gOne}{\mathbb{G}_1}
+\newcommand{\gTwo}{\mathbb{G}_2}
+\newcommand{\gT}{\mathbb{G}_T}
+\newcommand{\ee}{e \colon \mathbb{G}_1 \times \mathbb{G}_2 \rightarrow \mathbb{G}_T}
+\newcommand{\ev}{\mathrm{ev}}
+\newcommand{\cyclotomic}[2]{\Phi_{#1}(#2)}
+\newcommand{\fq}[1]{\mathbb{F}_{q^{#1}}}
+\newcommand{\cO}{\mathcal{O}}  
+\newcommand{\frob}{\pi_q}
+
+% Stuff to do
+\newcommand{\red}[1]{\color{red}#1\color{black}}
+
+% Languages
+\usepackage[english]{babel}
+
+% Math-related
+\usepackage{amsmath}
+\usepackage{amsfonts} % mathbb
+\usepackage{amsthm} % remark environment
+\usepackage{amssymb}
+
+% Hyperlinks
+\usepackage{hyperref}
+
+% Diagrams
+\usepackage{tikz-cd}
+
+% Comments
+\usepackage{verbatim}
+
+% Algorithms
+\usepackage{algorithm}
+\usepackage{algpseudocode}
+
+% Environments
+\theoremstyle{remark}
+\newtheorem{remark}{Remark}[section]
+
+\theoremstyle{plain}
+\newtheorem{lemma}{Lemma}[section]
+
+\title{Notes on bilinear pairings}
+\author{Federico Barbacovi \\ \href{mailto:f.barbacovi@nchain.com}{\texttt{f.barbacovi@nchain.com}}}
+
+
+\begin{document}
+
+\maketitle
+\tableofcontents
+
+\section{Introduction}
+
+These notes are meant to be a companion to the Github repository \href{https://github.com/nchain-innovation/zkscript_package}{zkscript}.
+In the notes we present the material which is turned into code in the repository.
+
+The notes are based on various references that are cited throughout.
+We also carry out some explicit calculations that we were not able to find in the literature.
+If you spot any mistake, please reach out.
```

Comment:
do we want a mail?

Reply:
I did put my email at the top of the document, do you think we should put another one? There's also the one in the README of the repository

Reply:
I was just thinking of writing reach out to f.barbacovi@nchain.com or contacts@nchain.com. But It is redundant